### PR TITLE
fix(DB/Loot) Use Condition for Quest Drop "Two Halves Become One"

### PR DESCRIPTION
**Pull request by LeFroid**
In quest Two Halves Become One, the Jabbering Ghoul drops an item (good luck other half charm) which is used to create the quest reward item (good luck charm). Because it is an indirect quest item drop, we must add a condition instead of relying on QuestRequired from creature_loot_template.

## Changes Proposed:
-  Use conditions table to reward good luck other-half-charm only if player has the requisite quest (5051)

## Issues Addressed:

- Closes 

## Tests Performed:

- 
- 

## How to Test the Changes:

1.  In game, enter command `.q add 5051` to add quest
2. Go to Jabbering ghoul, `.go c id 10801` 
3. Kill npc and confirm you can loot good luck other-half-charm (item id 12722)

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by Gultask** [comment]
I don't get it, the quest works normally for me without the PR. Why the change?

**Comment by LeFroid** [comment]
I thought the quest item wasn’t dropping when I last tried. Let me try again with a clean DB. This PR is unnecessary if the item is already dropping of course.

**Comment by Gultask** [comment]
No worries

**Comment by LeFroid** [comment]
Looks like this isn't needed afterall, thanks
